src/setup-elixir.js

```diff
@@ -1,97 +1,356 @@
 const core = require('@actions/core')
 const {exec} = require('@actions/exec')
-const {installElixir, installOTP} = require('./installer')
+const installer = require('./installer')
 const path = require('path')
 const semver = require('semver')
 const https = require('https')
 const {
-  fstat,
   promises: {readFile},
 } = require('fs')
 
-main().catch(err => {
+try {
+  main().catch((err) => {
+    core.setFailed(err.message)
+  })
+} catch (err) {
   core.setFailed(err.message)
-})
+}
```

Comment:
What is the `try {} catch {}` catching that `catch((err) => {})` doesn't catch?

Reply:
😄 exceptions, surely. I ran the tests without try/catch and get (UnhandledExceptionWarning). `.catch` catches stuff inside `async`, which is not the case for `checkPlatform`. You want me to change something here?

Reply:
In any case, more recently, I moved main from `async` to `Promise` which reverts this change.